File: sdd/sdd6/sbin/smb_preexec.pl
#!/usr/bin/perl
use Sys::Hostname;

$username = $ARGV[0];
# make sure that case is consistent for hostname...
# issue seen at Benham
$userhost = uc($ARGV[1]);
$should_allow = $ARGV[2];
$thishost = uc(hostname);

#open LOG, ">>/tmp/plog2" or die "couldn't open /tmp/plog";

$retval = 0;

# if username is rcud, and localhost, then always allow.
# if username is rcud, and not localhost, always deny. simple, eh?
if($username eq "rcud") {
    # always allow if this host
    if($userhost eq $thishost) {
	exit(0);
    } else { #always deny otherwise
	exit(1)
    }
}

#if not sharing, then don't allow anyone (already did RCUd check above)
if(($should_allow eq "no")) {
    $retval = 1;
}

#print LOG "user: $username host: $userhost this_host: $thishost retval: $retval\n";
#close $LOG;

exit($retval);

File: sdd/sdd5/sbin/smb_preexec.py
#!/usr/bin/python

import sys
import socket

if len(sys.argv) != 4:
    exit(1)

username = sys.argv[1]
userhost = sys.argv[2]
should_allow = sys.argv[3]
thishost = socket.gethostname()

# if username is rcud, and localhost, then always allow.
# if username is rcud, and not localhost, always deny. simple, eh?
if username == "rcud":
    # always allow if this host, do caseless comparison
    if userhost.lower() == thishost.lower():
        exit(0)
    else:
        exit(1)

#if not sharing, then don't allow anyone (already did RCUD check above)
if should_allow == "no":
    exit(1)

exit(0)
